validator-cli/src/ArbToEth/transactionHandler.ts

```diff
@@ -0,0 +1,191 @@
+import { VeaInboxArbToEth, VeaOutboxArbToEth } from "@kleros/vea-contracts/typechain-types";
+import { ClaimStruct } from "@kleros/vea-contracts/typechain-types/arbitrumToEth/VeaInboxArbToEth";
+import { JsonRpcProvider } from "@ethersproject/providers";
+import { messageExecutor } from "../utils/arbMsgExecutor";
+import { defaultEmitter } from "../utils/emitter";
+import { BotEvents } from "../utils/botEvents";
+import { ClaimNotSetError } from "../utils/errors";
+import { getBridgeConfig } from "../consts/bridgeRoutes";
+
+/**
+ * @file This file contains the logic for handling transactions from Arbitrum to Ethereum.
+ * It is responsible for:
+ *      challenge() - Challenge a claim on VeaOutbox(ETH).
+ *      withdrawChallengeDeposit() - Withdraw the challenge deposit.
+ *      sendSnapshot() - Send a snapshot from the VeaInbox(ARB) to the VeaOutox(ETH).
+ *      executeSnapshot() - Execute a sent snapshot to resolve dispute in VeaOutbox (ETH).
+ */
+
+type Transactions = {
+  challengeTxn: string | null;
+  withdrawChallengeDepositTxn: string | null;
+  sendSnapshotTxn: string | null;
+  executeSnapshotTxn: string | null;
+};
+
+export enum ContractType {
+  INBOX = "inbox",
+  OUTBOX = "outbox",
+}
+
+export class ArbToEthTransactionHandler {
+  public requiredConfirmations = 10;
+  public claim: ClaimStruct | null = null;
+  public chainId = 11155111;
+
+  public veaInbox: VeaInboxArbToEth;
+  public veaOutbox: VeaOutboxArbToEth;
+  public veaInboxProvider: JsonRpcProvider;
+  public veaOutboxProvider: JsonRpcProvider;
+  public epoch: number;
+  public emitter: typeof defaultEmitter;
+
+  public transactions: Transactions = {
+    challengeTxn: null,
+    withdrawChallengeDepositTxn: null,
+    sendSnapshotTxn: null,
+    executeSnapshotTxn: null,
+  };
+
+  constructor(
+    epoch: number,
+    veaInbox: VeaInboxArbToEth,
+    veaOutbox: VeaOutboxArbToEth,
+    veaInboxProvider: JsonRpcProvider,
+    veaOutboxProvider: JsonRpcProvider,
+    emitter: typeof defaultEmitter = defaultEmitter,
+    claim: ClaimStruct | null = null
+  ) {
+    this.epoch = epoch;
+    this.veaInbox = veaInbox;
+    this.veaOutbox = veaOutbox;
+    this.veaInboxProvider = veaInboxProvider;
+    this.veaOutboxProvider = veaOutboxProvider;
+    this.emitter = emitter;
+    this.claim = claim;
+  }
+
+  /**
+   * Check the status of a transaction.
+   *
+   * @param trnxHash Transaction hash to check the status of.
+   * @param contract Contract type to check the transaction status in.
+   *
+   * @returns False if transaction is pending || not final || not made, else True.
+   */
+  public async checkTransactionStatus(trnxHash: string | null, contract: ContractType): Promise<number> {
+    let provider: JsonRpcProvider;
+    if (contract === ContractType.INBOX) {
+      provider = this.veaInboxProvider;
+    } else if (contract === ContractType.OUTBOX) {
+      provider = this.veaOutboxProvider;
+    }
+
+    if (trnxHash == null) {
+      return 0;
```

Comment:
Can we provide an enum instead? I would have troubles trying to remember the meaning of a specific number, but the enum would give a tag to it.